warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -1897,14 +1897,49 @@ public static final GeoTimeSerie subSerie(GeoTimeSerie gts, long starttimestamp,
     // Extract values/locations/elevations that lie in the requested interval
     //
 
-    // We know how many data will the new GTS so we provision arrays to receive the data.
+    // We know how many data the new GTS will have so we provision arrays to receive the data.
     int count = lastidx - firstidx + 1;
-    GTSHelper.multiProvision(subgts, gts.type, count, count);
 
-    for (int i = firstidx; i <= lastidx; i++) {
-      setValue(subgts, gts.ticks[i], null != gts.locations ? gts.locations[i] : GeoTimeSerie.NO_LOCATION, null != gts.elevations ? gts.elevations[i] : GeoTimeSerie.NO_ELEVATION, valueAtIndex(gts, i), overwrite);
+    // Allocate locations/elevations arrays so they get expanded by the call to multiProvision
+    if (null != gts.locations) {
+      subgts.locations = new long[0];
+    }
+    if (null != gts.elevations) {
+      subgts.elevations = new long[0];
     }
 
+    GTSHelper.multiProvision(subgts, gts.type, count, count);
+
+    if (!overwrite) {
+      if (null != subgts.locations) {
+        System.arraycopy(gts.locations, firstidx, subgts.locations, 0, count);
+      }
+      if (null != subgts.elevations) {
+        System.arraycopy(gts.elevations, firstidx, subgts.elevations, 0, count);
+      }
+      if (null != subgts.ticks) {
+        System.arraycopy(gts.ticks, firstidx, subgts.ticks, 0, count);
+      }
+      if (null != subgts.longValues) {
+        System.arraycopy(gts.longValues, firstidx, subgts.longValues, 0, count);
+      }
+      if (null != subgts.doubleValues) {
+        System.arraycopy(gts.doubleValues, firstidx, subgts.doubleValues, 0, count);
+      }
+      if (null != subgts.stringValues) {
+        System.arraycopy(gts.stringValues, firstidx, subgts.stringValues, 0, count);
+      }
+      if (null != gts.booleanValues) {
+        subgts.booleanValues = gts.booleanValues.get(firstidx, lastidx + 1);
+      }
```

Comment:
The type of subgts is still UNDEFINED. It needs either to be set with each array copy, or by multiProvision if it does not affect it other usages.